package.json

```
@@ -1,15 +1,17 @@
 {
   "name": "react-native-website-monorepo",
   "private": true,
+  "engines": {
+    "node": ">=20"
+  },
   "workspaces": [
     "website",
     "plugins/*",
     "scripts/*",
     "sync-api-docs"
   ],
   "scripts": {
-    "postinstall": "yarn update-lock",
-    "update-lock": "yarn-deduplicate",
+    "update-lock": "yarn dedupe",
```

```suggestion
    "postinstall": "yarn update-lock",
    "update-lock": "yarn dedupe",
```

We still want to run dedupe on each install automatically, the fact that `dedupe` cannot be a part of `posinstall` is known, but besided proposed CI solution we need one for local checkout, see:
* https://github.com/yarnpkg/berry/issues/4976

---

i removed this because i had a memory leak issue where yarn install will spawn endless node processes. will revert and see if it happens again

---

If the `posinstall` approach fails after the upgrade we can try using an another lifecycle stage which makes sense, or write very simple yarn plugin which will run the command in `afterAllInstalled` hook, like suggested in the referenced above issue.

---

it seems calling yarn dedupe directly works, but invoking it from another script causes the infinite loop.

---

feel free to reproduce the bug on your machine

---

nevermind it works locally but breaks in ci, sigh

---

i created a [plugin](https://github.com/facebook/react-native-website/pull/4489/commits/ed83f09caa4d04b4c9c87ad2823683c8460c94d6) and it caused a freeze as well, i think it's because yarn dedupe runs install internally so i's stuck in an infinite loop

We should just get rid of it locally and make it run in ci only. if ci fails, then just run dedupe locally and push

---

It's fine approach short term, but it would make the development process a bit more cumbersome, since we validate the deduplication state on the CI. Would be nice to mention the required now dedupe step in contributing section.

But in long term, we should explore other options how to ensure that the lock is deduplicated automatically, when running locally.